clang/test/CXX/temp/temp.constr/temp.constr.order/class-template-partial-specializations.cpp

```diff
@@ -33,11 +33,12 @@ class D{}; // expected-note{{previous definition is here}}
 template<typename T>
 class D<T>{}; // expected-error{{class template partial specialization does not specialize any template argument; to define the primary template, remove the template argument list}} expected-error{{redefinition of 'D'}}
 
-template<typename T> requires C1<T> // expected-note{{previous template declaration is here}}
-class E{};
+template<typename T> requires C1<T>
+class E{}; // expected-note{{previous definition is here}}
 
-template<typename T> // expected-error{{requires clause differs in template redeclaration}}
+template<typename T>
```

Comment:
Although we error on the specialization of `E<T>`,  I still feel like the loss of the error `requires clause differs` is not so clear as the previous. @cor3ntin WDYT?

Reply:
Hmm, it looks like only we would diagnose an additional `requires clause differs` error.

https://gcc.godbolt.org/z/6doaeMEr3

Reply:
I can move the point at which we emit the redefinition error to be after the check for equivalent _template-heads_, if you'd like. Then we would get the "requires clause differs in template redeclaration" error instead.